src/dmd/json.d

```diff
@@ -857,17 +858,165 @@ public:
 extern (C++) void json_generate(OutBuffer* buf, Modules* modules)
 {
     scope ToJsonVisitor json = new ToJsonVisitor(buf);
-    json.arrayStart();
-    json.generateCompilerInfo();
-    json.generateBuildInfo();
-    for (size_t i = 0; i < modules.dim; i++)
-    {
-        Module m = (*modules)[i];
-        if (global.params.verbose)
-            message("json gen %s", m.toChars());
-        m.accept(json);
-    }
-    json.generateSemantics();
-    json.arrayEnd();
-    json.removeComma();
+
+    if (global.params.jsonQuery is null)
```

Comment:
```
if global.params.jsonQuery is null
  global.params.jsonQuery = "modules"; // or whatever we want default query to be; maybe put all the queries here

# then the rest can assume we always have `jsonQuery`
```

Reply:
Its definitely better to not break backwards compatibility.  If at some point in the future someone determines it's OK in this case, then it's simple to remove the special case.